spec/controllers/polls_controller_spec.rb

```diff
@@ -139,11 +184,33 @@
   describe 'GET #edit' do
     let(:poll) { FactoryGirl.create(:valid_poll) }
 
-    it 'should be success' do
+    it 'should be succes' do
       expect(get :edit, id: poll.id).to be_succes
     end
     it 'should render template edit' do
       expect(get :edit, id: poll.id).to render_template(:edit)
     end
   end
+
+  describe 'POST #make_choise' do
+    before(:each) { @poll = FactoryGirl.create(:valid_poll) }
+    let(:preferences) { @poll.options.ids.map { |id| 'option_' + id.to_s }.reverse }
+
+    it 'should be succes' do
+      expect(xhr :post, :make_choise, id: @poll.id, choise_array: preferences).to be_succes
+    end
+
+    it 'should save vote results if vote cast in first time' do
+      xhr :post, :make_choise, id: @poll.id, choise_array: preferences
+
+      expect { @poll.reload }.to change { @poll.vote_results.count }.by(1)
+    end
+
+    it 'should save vote results only once' do
+      xhr :post, :make_choise, id: @poll.id, choise_array: preferences
+      xhr :post, :make_choise, id: @poll.id, choise_array: preferences
+
+      expect { @poll.reload }.to change { @poll.vote_results.count }.by(1)
```

Comment:
This way is much more robust — the actual change is enclosed in the `{ }` block, so the accidental Rails model invalidation won't ruin your spec.